diff -c -b /home/tehom/.emacs.d/emacspeak/emacspeak-31.0/lisp/g-client/old-gblogger.el /home/tehom/.emacs.d/emacspeak/emacspeak-31.0/lisp/g-client/gblogger.el
*** /home/tehom/.emacs.d/emacspeak/emacspeak-31.0/lisp/g-client/old-gblogger.el	Wed Nov  5 17:45:43 2008
--- /home/tehom/.emacs.d/emacspeak/emacspeak-31.0/lisp/g-client/gblogger.el	Fri Feb 19 22:10:07 2010
***************
*** 91,96 ****
--- 91,102 ----
                  (string :tag "Save password in .emacs"))
    :group 'gblogger)
  
+ (defcustom gblogger-posting-url "http://invalid.com" 
+   "URL for posting to this post.
+ You can get the URL from a [post] link shown by `gblogger-blog'"
+   :type 'string
+   :group 'gblogger)
+ 
  (defcustom gblogger-author (user-full-name)
    "Author name under which we post."
    :type 'string
***************
*** 110,116 ****
    <title mode=\"escaped\" type=\"text/html\">%s </title>
    <content type='xhtml'>
      <div xmlns=\"http://www.w3.org/1999/xhtml\">
! <!--content goes here -->
      </div>
    </content>
  </entry>"
--- 116,122 ----
    <title mode=\"escaped\" type=\"text/html\">%s </title>
    <content type='xhtml'>
      <div xmlns=\"http://www.w3.org/1999/xhtml\">
! %s
      </div>
    </content>
  </entry>"
***************
*** 206,212 ****
     (substitute-command-keys "Use \\[g-app-publish] to publish your edits .")))
  
  ;;;###autoload
! (defun gblogger-new-entry (url)
    "Create a new Blog post."
    (interactive
     (list
--- 212,218 ----
     (substitute-command-keys "Use \\[g-app-publish] to publish your edits .")))
  
  ;;;###autoload
! (defun gblogger-new-entry (url &optional title text)
    "Create a new Blog post."
    (interactive
     (list
***************
*** 214,220 ****
    (declare (special gblogger-auth-handle gblogger-new-entry-template
                      gblogger-generator-name ))
    (g-auth-ensure-token gblogger-auth-handle)
!   (let* ((title (read-string "Title: "))
           (buffer (get-buffer-create title)))
      (save-excursion
        (set-buffer buffer)
--- 220,226 ----
    (declare (special gblogger-auth-handle gblogger-new-entry-template
                      gblogger-generator-name ))
    (g-auth-ensure-token gblogger-auth-handle)
!   (let* ((title (or title (read-string "Title: ")))
           (buffer (get-buffer-create title)))
      (save-excursion
        (set-buffer buffer)
***************
*** 227,236 ****
        (insert
         (format gblogger-new-entry-template
                 gblogger-generator-name gblogger-generator-name
!                gblogger-author title)))
      (switch-to-buffer buffer)
      (search-backward "<div" nil t)
      (forward-line 1)
      (message
       (substitute-command-keys
        "Use \\[g-app-publish] to publish your edits ."))))
--- 233,245 ----
        (insert
         (format gblogger-new-entry-template
                 gblogger-generator-name gblogger-generator-name
!                gblogger-author title
! 	       (or str "<!--content goes here -->")
! 	  )))
      (switch-to-buffer buffer)
      (search-backward "<div" nil t)
      (forward-line 1)
+     ;;(insert str)
      (message
       (substitute-command-keys
        "Use \\[g-app-publish] to publish your edits ."))))

Diff finished.  Fri Feb 19 22:18:25 2010