--- servers/espeak	2008-10-02 19:43:51.000000000 -0500
+++ servers/espeak	2009-05-19 11:16:25.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/tcl
+#!/usr/bin/tclsh
 # Keywords: Emacspeak, eSpeak , TCL
 # {{{ LCD Entry: 
 
@@ -34,6 +34,7 @@
 # }}}
 # {{{source common code 
 
+package require Tclx
 set wd [file dirname $argv0]
 source $wd/tts-lib.tcl
 
@@ -498,7 +499,7 @@
 signal ignore {sigint}
 
 # Set input encoding to utf-8
-fconfigure file0 -encoding utf-8
+fconfigure stdin -encoding utf-8
 
 #initialize eSpeak
 tts_initialize
