Index: emacspeak-org.el
===================================================================
--- emacspeak-org.el	(revision 8334)
+++ emacspeak-org.el	(working copy)
@@ -269,8 +269,8 @@
 
 (loop for f in
       '(
-        org-timestamp-down org-timestamp-down-day
-                           org-timestamp-up org-timestamp-up-day)
+        org-timestamp-down-day
+	org-timestamp-up-day)
       do
       (eval
        `(defadvice ,f (after emacspeak pre act comp)
@@ -279,6 +279,28 @@
             (emacspeak-auditory-icon 'select-object)
             (emacspeak-speak-line)))))
 
+(loop for f in
+      '(
+        org-timestamp-down org-timestamp-up)
+      do
+      (eval
+       `(defadvice ,f (around emacspeak pre act comp)
+	  "Provide auditory feedback."
+	  (when (ems-interactive-p )
+	    (emacspeak-auditory-icon 'select-object)
+	    (let ((prior 
+		   (append (buffer-substring 
+			    (save-excursion (beginning-of-line) (point))
+			    (save-excursion (end-of-line) (point))
+			    ) nil)))
+	      ad-do-it
+	      (save-excursion 
+		(beginning-of-line)
+		(while (equal (char-after) (pop prior))
+		  (forward-char))
+		(skip-chars-backward "0-9")
+		(emacspeak-speak-line 1)))))))
+
 (defadvice org-eval-in-calendar (after emacspeak pre act comp)
   "Speak what is returned."
   (declare (special org-ans2))
