diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index 6362473..2bbf464 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -1008,8 +1008,9 @@ Produce an auditory icon if possible."
   "Make comint speak its output."
   (let ((inhibit-read-only t)
         (monitor emacspeak-comint-output-monitor)
+		(buffer (process-buffer (ad-get-arg 0)))
         (dtk-stop-immediately nil))
-    (set-buffer (process-buffer (ad-get-arg 0)))
+    (set-buffer buffer)
     ad-do-it
     (when (and (boundp 'comint-last-prompt-overlay)
                comint-last-prompt-overlay)
@@ -1022,9 +1023,7 @@ Produce an auditory icon if possible."
         'rear-sticky nil)))
     (when (and
            (or emacspeak-comint-autospeak emacspeak-speak-comint-output)
-           (or monitor
-               (eq (selected-window)
-                   (get-buffer-window (process-buffer (ad-get-arg 0))))))
+           (or monitor (eq (window-buffer) buffer)))
       (emacspeak-speak-region comint-last-output-start (point )))
     ad-return-value))
 
