Index: emacspeak-org.el
===================================================================
--- emacspeak-org.el	(revision 8872)
+++ emacspeak-org.el	(working copy)
@@ -167,11 +167,10 @@
           (when (ems-interactive-p )
             (cond
              ((org-at-table-p 'any)
-              (let ((emacspeak-show-point t))
-                (skip-syntax-forward " ")
-                (emacspeak-speak-line)
-                (emacspeak-auditory-icon 'large-movement)))
-             (t (emacspeak-auditory-icon 'select-object)))))))
+              (emacspeak-org-table-speak-current-element))
+             (t
+	      (emacspeak-speak-line)
+	      (emacspeak-auditory-icon 'select-object)))))))xrjp
 
 (defadvice org-overview (after emacspeak pre act comp)
   "Provide auditory feedback."
@@ -442,7 +441,12 @@
 (defadvice org-return (after emacspeak pre act comp)
   "Provide auditory feedback."
   (when (ems-interactive-p )
-    (emacspeak-speak-line)))
+    (cond
+     ((org-at-table-p 'any)
+      (emacspeak-org-table-speak-current-element))
+     (t
+      (emacspeak-speak-line)
+      (emacspeak-auditory-icon 'select-object)))))
 
 ;;}}}
 ;;{{{ mode hook:
@@ -535,6 +539,18 @@
   "Provide auditory feedback."
   (emacspeak-auditory-icon 'close-object))
 
+;;;###autoload
+(defun emacspeak-org-table-speak-current-element ()
+  "echoes current table element"
+  (interactive)
+  (dtk-speak-and-echo (org-table-get-field)))
+
+
+(defadvice org-table-next-field (after emacspeak pre act comp)
+  "Provide auditory feedback."
+  (emacspeak-org-table-speak-current-element))
+
+
 ;;}}}
 (provide 'emacspeak-org)
 ;;{{{ end of file
